File: BEmuJava/bemu/src/main/java/com/bemu/BEmu/IntradayBarRequest/ElementIntradayBarString.java
package com.bemu.BEmu.IntradayBarRequest;

import com.bemu.BEmu.Name;
import com.bemu.BEmu.Schema;
import com.bemu.BEmu.Element;

public class ElementIntradayBarString extends Element
{
    private final String _value;
    private final String _name;

    ElementIntradayBarString(String name, String value)
    {
        this._name = name;
        this._value = value;
    }
	
    public Name name()
    {
    	return new Name(this._name);
    }
    
    public int numValues()
    {
    	return 1;
    }
    
    public int numElements()
    {
    	return 0;
    }
    
    public boolean isComplexType()
    {
    	return false;
    }
    
    public boolean isArray()
    {
    	return false;
    }
    
    public boolean isNull()
    {
    	return false;
    }
    
    public Schema.Datatype datatype()
    {
    	return Schema.Datatype.STRING;
    }
    
    protected StringBuilder prettyPrint(int tabIndent)
    {
    	return super.prettyPrintHelper(tabIndent, String.valueOf(this._value));
    }
    
    public Object getValue()
    {
    	return this._value;
    }
    
    public String getValueAsString()
    {
    	return this._value;
    }

}

File: BEmu_cpp/src/IntradayBarRequest/ElementIntradayBarString.cpp
//------------------------------------------------------------------------------------------------
// <copyright project="BEmu_cpp" file="src/IntradayBarRequest/ElementIntradayBarString.cpp" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------------------------

#include "IntradayBarRequest/ElementIntradayBarString.h"
#include "BloombergTypes/Name.h"
#include "BloombergTypes/ElementPtr.h"

namespace BEmu
{
	namespace IntradayBarRequest
	{
		ElementIntradayBarString::ElementIntradayBarString(const std::string& name, const std::string& value)
		{
			this->_name = name;
			this->_value = value;
		}

		ElementIntradayBarString::~ElementIntradayBarString()
		{
		}

		Name ElementIntradayBarString::name() const
		{
			Name result(this->_name.c_str());
			return result;
		}

		int ElementIntradayBarString::numValues() const
		{
			return 1;
		}

		int ElementIntradayBarString::numElements() const
		{
			return 0;
		}

		bool ElementIntradayBarString::isArray() const
		{
			return false;
		}

		bool ElementIntradayBarString::isComplexType() const
		{
			return false;
		}

		char* ElementIntradayBarString::getValueAsString(int index) const
		{
			if(index == 0)
				return ElementPtr::toCharPointer(this->_value);
			else
				throw elementPtrEx;
		}

		std::ostream& ElementIntradayBarString::print(std::ostream& stream, int level, int spacesPerLevel) const
		{
			this->prettyPrintHelper(stream, level, spacesPerLevel, this->_value);
			return stream;
		}



	}
}